prboom2/src/SDL/i_system.c

```diff
@@ -285,24 +285,41 @@ const char* I_GetTempDir(void)
 }
 
 #else
-// cph - V.Aguilar (5/30/99) suggested return ~/.lxdoom/, creating
-//  if non-existant
-// cph 2006/07/23 - give prboom+ its own dir
-static const char prboom_dir[] = {"/.dsda-doom"}; // Mead rem extra slash 8/21/03
+
+static char *I_ConcatDir(const char *var, const char *subdir)
+{
+  static char buffer[PATH_MAX];
```

Comment:
It's too dangerous to have this as a static, in case we reuse this function later on. Let's allocate the result using the string lengths of the args.